packages/shopping-cart/src/create-request-cart-product.ts

```diff
@@ -3,7 +3,7 @@ import type { RequestCartProduct, MinimalRequestCartProduct } from './types';
 export default function createRequestCartProduct(
 	properties: MinimalRequestCartProduct
 ): RequestCartProduct {
-	if ( ! properties.product_slug ) {
+	if ( ! properties.product_slug || properties.product_slug === 'undefined' ) {
 		throw new Error( 'product_slug is required for request cart products' );
```

Comment:
This error message is throwing my brain for a loop. I think it is because I am not super familiar with how the cart is generated. To me, it reads as either missing a preposition or a tense issue. 

Could we be more explicit about which function this error is coming from or that this is one step of a series?

Maybe, `product_slug is required for createRequestCartProduct` or `product_slug is required for requesting products to generate a cart`

Reply:
That's fair. If you don't know what a "request cart" is, this doesn't make a lot of sense. (FWIW this is explained [in the README](https://github.com/Automattic/wp-calypso/blob/8611a88fdad17155ff725b4fc0ec7631f2b56936/packages/shopping-cart/README.md#L1) for the shopping-cart package although even there it's a bit technical. A "request cart" is the cart object used to request a shopping cart update, and a "response cart" is what you get back.) I'll update this to have an error that's easier to read.